### PR TITLE
Remove occurences of $$ in arm_cs esil ##refactor

### PR DESCRIPTION
**Pull request by condret**
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/anal/p/anal_arm_cs.c [hunk 1]
```diff
@@ -2439,7 +2439,7 @@ static int analop_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len
 		r_strbuf_appendf (&op->esil, "%s,!,?{,32,%s,=,BREAK,},0,%s,=,%s,%s,<<,0x80000000,&,!,?{,1,%s,+=,11,GOTO,}", REG (1), REG (0), REG (0), REG (0), REG (1), REG (0));
 		break;
 	case ARM_INS_IT:
-		r_strbuf_appendf (&op->esil, "2,$$,+,pc,=");
+		r_strbuf_appendf (&op->esil, "2,pc,+,%d,pc,-,pc,:=", op->size);
```
Comment (trufae) [inline libr/anal/p/anal_arm_cs.c:2442]: i dont think pc is the same as $$

Comment (condret) [reply libr/anal/p/anal_arm_cs.c:2442]: yes, I found that out, pc is $$ + op->size.

Comment (trufae) [reply libr/anal/p/anal_arm_cs.c]: the register is updated before stepping

Comment (trufae) [reply libr/anal/p/anal_arm_cs.c]: But that behaviour changes depending on thumb, arm and the alignment if its 2 or 4. We have tests for this madness and thats why it was using $$

Comment (trufae) [reply libr/anal/p/anal_arm_cs.c]: Because you cant easily compute that from pc

Comment (condret) [reply libr/anal/p/anal_arm_cs.c]: why is this behavior changed based on thumb? anyways, I need to deprecate $$ for esil_dfg

Comment (aemmitt-ns) [reply libr/anal/p/anal_arm_cs.c]: op->size is right regardless of whether its thumb right?

Comment (trufae) [reply libr/anal/p/anal_arm_cs.c]: I think that we had enough tests to confirm that the behaviour was correct. We can do a blind guess and merge and hopefully we didnt missed any test. Sgty?

**Pull request merged**